tests/SCsub

```diff
@@ -7,7 +7,8 @@ env.tests_sources = []
 env_tests = env.Clone()
 
 # Include GDNative headers.
-env_tests.Append(CPPPATH=["#modules/gdnative/include"])
+if env["module_gdnative_enabled"]:
+    env_tests.Append(CPPPATH=["#modules/gdnative/include"])
```

Comment:
Shouldn't the GDNative module be responsible for registering its own tests?

Reply:
Yes, but then again the way tests are collected for modules have global meaning, and if you do define the same include paths, it breaks all other paths (including GDNative core includes themselves), I'm not confident to modify those and that's probably not needed/required (may as well break compatibility with other GDNative plugins outside of Godot I guess).

Reply:
Using relative paths may be a solution (`#include "../include/gdnative/string.h"`) within `modules/gdnative/tests/test_string.h`, if you deem this acceptable.

Reply:
Then yes, use a relative path in the test itself.

Reply:
Just tried that now, similar errors:
```
[100%] D:\src\godot\4\modules\gdnative\tests\../include/gdnative/string.h(74): fatal error C1083: Cannot open include file: 'gdnative/array.h': No such file or directory 
scons: *** [tests\test_main.windows.tools.64.obj] Error 2
```

https://github.com/godotengine/godot/blob/4b846ebc56a0493e31a2767a91816cdf5d5b2c9e/modules/gdnative/include/gdnative/string.h#L74-L76

Reply:
Currently, all modules tests are just collected and included as part of `tests/test_main.cpp`, when you ask whether GDNative is responsible for registering tests, the answer is: no. 🙂

Reply:
Alright, it's not worse than it was anyway so let's merge. Thanks for looking into options.

Eventually we should make this more modular, with a system where each module is responsible for registering its tests.